hubploy/auth.py

```diff
@@ -53,17 +54,20 @@ def registry_auth_aws(deployment, project, service_key):
 
     This changes *global machine state* on where docker can push to!
     """
-    service_key_path = os.path.join(
-        'deployments', deployment, 'secrets', service_key
+    # This assumes that AWS_ACCESS_KEY_ID and AWS_SECRET_KEY_ID are in env vars
```

Comment:
We should keep these in secrets/ for sure, and not have them be external environment variables. In deployment repos, we want to have only one secret - the key to unlock all our secret/ files. We should avoid depending on extra environment variables.

Reply:
Right. So the gcloud service key has the following structure:

```json
{
  "type": "service_account",
  "project_id": "myproject",
  "private_key_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
  "private_key": "-----BEGIN PRIVATE KEY-----\nXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\n-----END PRIVATE KEY-----\n",
  "client_email": "mykey@myproject.iam.gserviceaccount.com",
  "client_id": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/mykey%40myproject.iam.gserviceaccount.com"
}

Does AWS have a similar key file format? If not, should we roll our own?
```